nixos/doc/manual/installation/installing-kexec.section.md

```diff
@@ -0,0 +1,59 @@
+# "Booting" into NixOS via kexec (PXE) {#sec-booting-via-kexec}
+
+In some cases, your system might already be booted into/preinstalled with
+another Linux distribution, and booting NixOS by attaching an installation
+image is quite a manual process.
+
+This is particularly useful for (cloud) providers where you can't boot a custom
+image, but get some Debian or Ubuntu installation.
+
+In these cases, it might be easier to use `kexec` to "jump into NixOS" from the
+running system, which only assumes `bash` and `kexec` to be installed on the
+machine.
+
+Note that kexec may not work correctly on some hardware, as devices are not
+fully re-initialized in the process. In practice, this however is rarely the
+case.
+
+To build the necessary files from your current version of nixpkgs,
+you can run:
+
+```ShellSession
+nix-build -A kexec.x86_64-linux '<nixpkgs/nixos/release.nix>'
+```
+
+This will create a `result` directory containing the following:
+ - `bzImage` (the Linux kernel)
+ - `initrd` (the initrd file)
+ - `kexec-boot` (a shellscript invoking `kexec`)
+
+These three files are meant to be copied over to the other already running
+Linux Distribution.
+
+Note it's symlinks pointing elsewhere, so `cd` in, and use
+`scp * root@$destination` to copy it over, rather than rsync.
+
+Once you finished copying, execute `kexec-boot` *on the destination*, and after
+some seconds, the machine should be booting into an (ephemeral) NixOS
+installation medium.
+
+Note this is using `makeKexec` under the hood, which, similar to `makeNetboot`,
+allows passing in a custom system config via the `module` argument.
+
+Instead of using the default installer image, you can also pass in a custom
+configuration:
+
+```nix
+
+{ system ? builtins.currentSystem }:
+
+{
+  myNetboot = makeKexec {
+    module = ./path/to/my/config.nix;
+    inherit system;
+  };
+}
+```
```

Comment:
+1 for adding to `release.nix`, so we have ready-built default images, but why not keep the module? 

`makeKexec` is a let binding, which is only accessible by Nixpkgs forks.

You could use the module in `release.nix`. That will keep `release.nix` simple and serve as a reasonable example of how to build one in an expression, for people who prefer to dive into code and learn by example.

I guess the alternative is to expose `makeKexec`, but that would create yet another entrypoint into NixOS, which is hostile to composition.

This seems to be a working example with the module:

```nix
let
  kexecConfig = import <nixpkgs/nixos> {
    configuration = { modulesPath, ... }: {
      imports = [
        (modulesPath + "/installer/kexec/kexec-boot.nix")
        (modulesPath + "/installer/netboot/netboot-minimal.nix")
        # for example
        ./configuration.nix
      ];
    };
  };
in
kexecConfig.config.system.build.kexecBoot
```

or perhaps a command-line alternative

`configuration.nix`
```
{ modulesPath, ... }: {
  imports = [
    (modulesPath + "/installer/kexec/kexec-boot.nix")
    (modulesPath + "/installer/netboot/netboot-minimal.nix")
  ];
};
```

```
nix-build '<nixpkgs/nixos>' \
  --arg configuration ./configuration.nix \
  --attr config.system.build.kexecBoot
```

Reply:
It is still possible to provide a simplified interface, as previously discussed in https://github.com/NixOS/nixpkgs/pull/161253#discussion_r811306610

I've been somewhat critical of it because of eval performance, but the simplicity of such an interface is very much worth it for use cases where the kexec variant isn't evaluated often. https://github.com/NixOS/rfcs/pull/22 would make it a no-brainer.